chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/UpdateListingStatusJob.java

```diff
@@ -0,0 +1,163 @@
+package gov.healthit.chpl.scheduler.job;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.context.support.SpringBeanAutowiringSupport;
+
+import gov.healthit.chpl.auth.permission.GrantedPermission;
+import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
+import gov.healthit.chpl.domain.CertificationStatus;
+import gov.healthit.chpl.domain.CertificationStatusEvent;
+import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
+import gov.healthit.chpl.domain.ListingUpdateRequest;
+import gov.healthit.chpl.exception.ValidationException;
+import gov.healthit.chpl.manager.CertifiedProductDetailsManager;
+import gov.healthit.chpl.manager.CertifiedProductManager;
+
+public class UpdateListingStatusJob extends QuartzJob {
+    private static final Logger LOGGER = LogManager.getLogger("updateListingStatusJobLogger");
+
+    @Autowired
+    private CertifiedProductDetailsManager certifiedProductDetailsManager;
+
+    @Autowired
+    private CertifiedProductManager certifiedProductManager;
```

Comment:
Some more field injection that might be able to be moved to a constructor.

Reply:
I don't think we can use constructor injection here.  The quartz job actually "injects" the fields after the job has started running, using:
`SpringBeanAutowiringSupport.processInjectionBasedOnCurrentContext(this);`

Reply:
Yeah, the same comment was made on my Quartz job recently. I wasn't able to do that there either

Reply:
ahh ok. cool